wordpress-mobile/WordPress-iOS
# Issue/7371 crash on media picker

Fixes #7371 

This PR sorts the crashing issue by two main things:
 - reduce the amount of reload of the data that was being done ( there was some bugs on the navigation picker code that made the data to be reloaded two or three times on load)
 -  and when a group is changed (ex:switch to media library) don't load the data was a set of change but instead do a full reload

To test:
 - Install the app fresh
 - Select a blog
 - Open the media picker by trying to change the site icon or creating a new post
 - Switch to the WordPress Media Library
 - Check that no crash happen
 - Switch to another blog do the same and see if no crash happen
 - Check if picker is still working well when selecting things.

If all goes well I will need to release a new version of the MediaPicker pod and then update the Podfile to a proper version. DO NOT MERGE until I change that.

Needs review: @kurzee and @elibud
